src/switch_ivr_async.c

```diff
@@ -4786,10 +4784,7 @@ static void *SWITCH_THREAD_FUNC speech_thread(switch_thread_t *thread, void *obj
 				switch_channel_event_set_data(channel, dup);
 				switch_event_fire(&dup);
 			}
-
-		}
-
-		if (switch_core_session_queue_event(sth->session, &event) != SWITCH_STATUS_SUCCESS) {
+		} else if (switch_core_session_queue_event(sth->session, &event) != SWITCH_STATUS_SUCCESS) {
```

Comment:
I got a concern that change of behaviour of how SWITCH_ASR_FLAG_FIRE_EVENTS works may cause some issues. This commit changes it cause originally if SWITCH_ASR_FLAG_FIRE_EVENTS is set then it would additionally fire event and always queue it, but after this change it would only fire it. So, maybe instead of changing how SWITCH_ASR_FLAG_FIRE_EVENTS works, we would add new variable which would indicate event should only be fired and not queued. If that new flag is set then only fire event, do not queue it, if this variable is not set then code would be unchanged. Var could be name asr-events-fire-only or asr-events-do-not-queue, etc.

Reply:
Good point. I made some approach here as I follow the implementation on ASR. Variable are set in `switch_ivr_detect_speech_init` or `switch_ivr_detect_speech` then introduce new ASR Flag `SWITCH_ASR_FLAG_QUEUE_EVENTS` to enable/disable ASR event queue (default is enabled)